--- a/package/boot/uboot-rockchip/Makefile
+++ b/package/boot/uboot-rockchip/Makefile
@@ -131,6 +131,13 @@ endef
     friendlyarm_nanopc-t4
 endef
 
+define U-Boot/nanopi-m4-rk3399
+  $(U-Boot/rk3399/Default)
+  NAME:=NanoPi M4
+  BUILD_DEVICES:= \
+    friendlyarm_nanopi-m4
+endef
+
 define U-Boot/nanopi-r4s-rk3399
   $(U-Boot/rk3399/Default)
   NAME:=NanoPi R4S
@@ -445,5 +445,6 @@ UBOOT_TARGETS := \
   guangmiao-g4c-rk3399 \
   nanopc-t4-rk3399 \
+  nanopi-m4-rk3399 \
   nanopi-r4s-rk3399 \
   nanopi-r4se-rk3399 \
   rock-4c-plus-rk3399 \
